docs/source/essentials/data.md

```diff
@@ -2,108 +2,67 @@
 
-To provide a `context` to your resolvers, add an object to the Apollo Server constructor. For specific examples, follow the [backend/]() instructions.
+For specific examples, follow the [backend]() instructions.
 
 ```
-  const server = new ApolloServer(req => ({
-    typeDefs,
-    resolvers,
-    context: {
-      authScope: getScope(req)
-    }
-  }));
-
-  //resolver
-  (parent, _, context) => {
-    if(context.authScope !== ADMIN) throw AuthenticationError('not admin');
-    ...
+const server = new ApolloServer(req => ({
+  typeDefs,
+  resolvers,
+  context: {
+    authScope: getScope(req.headers.authorization)
   }
+}));
+
+// resolver
+(parent, _, context) => {
+  if(context.authScope !== ADMIN) throw AuthenticationError('not admin');
+  ...
+}
 ```
 
-The context can also be created asynchronous, allowing database connections and other operations to complete.
+The context can also be created asynchronously, allowing database connections and other operations to complete.
 
 ```
 context: async () => ({
   db: await client.connect(),
 })
 
-//resolver
+// resolver
 (parent, _, context) => {
   return context.db.query('SELECT * FROM table_name');
 }
 ```
 
-## Implementing Queries in Apollo Server
-
-Now that we understand the Query type, GraphQL types, and resolvers, we can explain the following code to define our schema and resolvers. This example shows the
-
-```js
-const { ApolloServer } = require('apollo-server');
-
-const typeDefs = `
-  type Process {
-    params: [String]
-    program: String
-  }
-
-  type Query {
-    process: Process
-    argv: [String]
-  }
-`;
-
-// Resolvers define the technique for fetching the types in the
-// schema.  We'll retrieve books from the "books" array above.
-const resolvers = {
-  Process: {
-    params: (parent) => parent.argv.slice(1)
-    program: (parent) => parent.argv[0]
-    url: (_,_,context) => context.req.baseUrl
-  }
-
-  Query: {
-    process: () => process
-    argv: () => process.argv
-  },
-};
-
-new ApolloServer({ typeDefs, resolvers, context: { req } })
-  .listen()
-  .then(({ url }) => {
-    console.log(`Visit ${url} to run queries!`);
-  });
-```
 
 ## Material Summary
 
-* The context is shared by all resolvers during a single query or mutation
-* setup the context on every request
-* Great place to store authentication (but this will be covered in authentication)
-* database connections
-  * mongo
-  * postgress
-  * fetch functions
-* N+1 problem
-  * A consideration for dataloader.
-* Here's how you add it to Apollo Server
-* State
-* BEST practice: keep your context code the same regardless of query/mutation that is coming in
+- [x] The context is shared by all resolvers during a single query or mutation
+- [x] setup the context on every request
+- [x] Great place to store authentication (but this will be covered in authentication)
+- [ ] database connections
+  - [ ] mongo
+  - [ ] postgress
+  - [ ] fetch functions
+- [x] N+1 problem
```

Comment:
should we add more about this, or will linking it to the best practices page like I did be enough?